plugins/event_source/sqs.py

```diff
@@ -0,0 +1,73 @@
+"""
+sqs.py
+
+An ansible-rulebook event source plugin for receiving events via an AWS SQS queue.
+
+Arguments:
+    region: AWS region containing the SQS queue.
+    queue:  Name of the SQS queue.
+    wait_seconds: The SQS long polling duration. Set to 0 to disable. Defaults to 2.
+"""
+
+import asyncio
+import logging
+import sys
+
+from typing import Any, Dict
+
+import botocore.exceptions
+
+from aiobotocore.session import get_session
+
+
+async def main(queue: asyncio.Queue, args: Dict[str, Any]):
+    logger = logging.getLogger()
+
+    region = args.get("region")
+    queue_name = args.get("queue")
+    wait_seconds = int(args.get("wait_seconds", 2))
+
+    # Boto should get credentials from ~/.aws/credentials or the environment
+    session = get_session()
+    async with session.create_client("sqs", region_name=region) as client:
+        try:
+            response = await client.get_queue_url(QueueName=queue_name)
+        except botocore.exceptions.ClientError as err:
+            if (
+                err.response["Error"]["Code"]
+                == "AWS.SimpleQueueService.NonExistentQueue"
+            ):
+                raise RuntimeError("Queue %s does not exist" % queue_name)
+            else:
+                raise
+
+        queue_url = response["QueueUrl"]
+
+        while True:
+            # This loop wont spin really fast as there is
+            # essentially a sleep in the receive_message call
+            response = await client.receive_message(
+                QueueUrl=queue_url,
+                WaitTimeSeconds=wait_seconds,
+            )
+
+            if "Messages" in response:
+                for msg in response["Messages"]:
+                    await queue.put(msg)
```

Comment:
What is the type of msg? It needs to be converted to dict if not already.

Reply:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sqs.html#SQS.Client.receive_message

<img width="732" alt="image" src="https://user-images.githubusercontent.com/773186/217622835-ede3f54a-d2b8-4bde-8d77-a07bc72eb20c.png">

Does EDA do type checking when pulling stuff off the queue? I can add extra guardrails here but seems like it could be overkill.

Reply:
This is good